mlflow/utils/autologging_utils/safety.py

```diff
@@ -820,6 +919,37 @@ def _validate_new_input(inp):
                 "such exception safe functions / classes.".format(inp)
             )
 
+    def _assert_same_type(autologging_call_input, user_call_input):
+        assert type(autologging_call_input) == type(
+            user_call_input
+        ), "Type of input to original function '{}' does not match expected type '{}'".format(
+            type(autologging_call_input), type(user_call_input)
+        )
+
+    def _assert_same_input_length(autologging_call_input, user_call_input):
+        length_difference = len(autologging_call_input) - len(user_call_input)
+        assert (
+            length_difference >= 0
+        ), "{} expected inputs are missing from the call to the original function.".format(
+            length_difference
+        )
+
+    def _assert_same_key_set(autologging_call_input, user_call_input):
```

Comment:
This method name isn't quite accurate. The method is asserting that the autologging call input kwarg keys are a superset of the user call input kwarg keys to make sure that we didn't delete user keyword arguments.